browser/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,472 @@
+    }
+  }
+
+  return false;
+}
+
+}  // namespace
+
+
+BraveCosmeticResourcesTabHelper::BraveCosmeticResourcesTabHelper(
+    content::WebContents* contents)
+    : WebContentsObserver(contents),
+    enabled_1st_party_cf_filtering_(false) {
+  if (BraveCosmeticResourcesTabHelper::observing_script_->empty()) {
+    *BraveCosmeticResourcesTabHelper::observing_script_ =
+        LoadDataResource(kCosmeticFiltersGenerated[0].value);
+  }
+}
+
+BraveCosmeticResourcesTabHelper::~BraveCosmeticResourcesTabHelper() {
+}
+
+void BraveCosmeticResourcesTabHelper::GetUrlCosmeticResourcesOnUI(
+    content::GlobalFrameRoutingId frame_id, const std::string& url,
+    bool do_non_scriplets, std::unique_ptr<base::ListValue> resources) {
+  if (!resources) {
+    return;
+  }
+  for (auto i = resources->GetList().begin();
+      i < resources->GetList().end(); i++) {
+    base::DictionaryValue* resources_dict;
+    if (!i->GetAsDictionary(&resources_dict)) {
+      continue;
+    }
+    std::string pre_init_script;
+    std::string scriplet_init_script = "";
+    std::string non_scriplet_init_script = "";
+    std::string json_to_inject;
+    if (base::JSONWriter::Write(
+        *(resources_dict->FindPath("injected_script")), &json_to_inject) &&
+            json_to_inject.length() > 1) {
+      scriplet_init_script = base::StringPrintf(k_scriplet_init_script,
+          json_to_inject.c_str());
+    }
+    if (do_non_scriplets) {
+      Profile* profile = Profile::FromBrowserContext(
+          web_contents()->GetBrowserContext());
+      enabled_1st_party_cf_filtering_ =
+          brave_shields::IsFirstPartyCosmeticFilteringEnabled(
+              HostContentSettingsMapFactory::GetForProfile(profile),
+                  GURL(url));
+      bool generichide = false;
+      resources_dict->GetBoolean("generichide", &generichide);
+      non_scriplet_init_script =
+          base::StringPrintf(k_non_scriplet_init_script,
+              enabled_1st_party_cf_filtering_ ? "true" : "false",
+              generichide ? "true" : "false");
+    }
+    pre_init_script = base::StringPrintf(k_pre_init_script,
+        scriplet_init_script.c_str(), non_scriplet_init_script.c_str());
+    auto* frame_host = content::RenderFrameHost::FromID(frame_id);
+    if (!frame_host)
+      return;
+    frame_host->ExecuteJavaScriptInIsolatedWorld(
+      base::UTF8ToUTF16(pre_init_script),
+      base::NullCallback(), ISOLATED_WORLD_ID_CHROME_INTERNAL);
+    if (do_non_scriplets) {
+      frame_host->ExecuteJavaScriptInIsolatedWorld(
+        base::UTF8ToUTF16(*BraveCosmeticResourcesTabHelper::observing_script_),
+        base::NullCallback(), ISOLATED_WORLD_ID_CHROME_INTERNAL);
+    }
+    // Working on css rules, we do that on a main frame only
+    if (!do_non_scriplets)
+      return;
+    CSSRulesRoutine(url, resources_dict, frame_id);
+  }
+}
+
+void BraveCosmeticResourcesTabHelper::CSSRulesRoutine(
+    const std::string& url_string, base::DictionaryValue* resources_dict,
```

Comment:
`const base::DictionaryValue*` ?

Reply:
it fails on GetList in that case as the method is not marked as const there
`candidate function not viable: 'this' argument has type 'const base::DictionaryValue', but method is not marked const`